File: backend/constants/roles.py
"""Rôles disponibles pour un chatbot.
La clé (ex: "support_client") est stockée en base et utilisée dans l'API.
Le label est uniquement pour l'affichage frontend.
"""

CHATBOT_ROLES = {
    "assistant": "Assistant virtuel",
    "support_client": "Support client",
    "commercial": "Assistant commercial",
    "ecommerce": "Conseiller e-commerce",
    "rh": "Assistant RH",
    "administratif": "Assistant administratif",
    "technique": "Assistant technique",
    "education": "Assistant pédagogique",
    "marketing": "Assistant marketing",
    "analytique": "Assistant analytique",
    "faq": "FAQ",
    "expert": "Expert métier",
    "reservation": "Assistant réservation",
    "community_manager": "Community Manager",
}

DEFAULT_ROLE = "assistant"
ALLOWED_ROLES = list(CHATBOT_ROLES.keys())

File: frontend/constants/roles.ts
export const CHATBOT_ROLES = [
  { value: "assistant", label: "Assistant virtuel" },
  { value: "support_client", label: "Support client" },
  { value: "commercial", label: "Assistant commercial" },
  { value: "ecommerce", label: "Conseiller e-commerce" },
  { value: "rh", label: "Assistant RH" },
  { value: "administratif", label: "Assistant administratif" },
  { value: "technique", label: "Assistant technique" },
  { value: "education", label: "Assistant pédagogique" },
  { value: "marketing", label: "Assistant marketing" },
  { value: "analytique", label: "Assistant analytique" },
  { value: "faq", label: "FAQ" },
  { value: "expert", label: "Expert métier" },
  { value: "reservation", label: "Assistant réservation" },
  { value: "community_manager", label: "Community Manager" },
] as const;

export const DEFAULT_ROLE = "assistant";

export function getRoleLabel(value: string | undefined | null): string {
  if (!value) return CHATBOT_ROLES[0].label;
  return CHATBOT_ROLES.find((r) => r.value === value)?.label ?? value;
}